tools/pkg/config/config.go

```diff
@@ -57,5 +61,37 @@ func Parse(fileName string) (Config, error) {
 		return Config{}, fmt.Errorf("failed to unmarshal as JSON: %w", err)
 	}
 
+	for _, image := range config.Images {
+		if err := image.SetDefaults(); err != nil {
+			return Config{}, fmt.Errorf("failed to set defaults for image %q: %w", image, err)
+		}
+	}
+
 	return config, nil
 }
+
+func (image *Image) SetDefaults() error {
+	parts := strings.Split(image.SourceImage, "/")
+	if len(parts) < 2 {
+		return fmt.Errorf("source image split into %d parts (>=2 parts expected)", len(parts))
+	}
+	repoName := parts[len(parts)-2]
+	imageName := parts[len(parts)-1]
+	image.defaultTargetImageName = "mirrored-" + repoName + "-" + imageName
+	return nil
+}
+
+func (image Image) TargetImageName() string {
```

Comment:
I think this should be `func (image *Image)` to not potentially cause confusion by mixing methods with pointer and value receivers

Reply:
This always trips me up. https://go.dev/doc/faq#methods_on_values_or_pointers says:

> Next is consistency. If some of the methods of the type must have pointer receivers, the rest should too, so the method set is consistent regardless of how the type is used.

Which is good enough for me :+1: